Sofa/Component/LinearSolver/Direct/tests/SparseLDLSolver_test.cpp

```diff
@@ -178,3 +181,50 @@ TEST(SparseLDLSolver, MatrixFactorization)
     static const sofa::type::vector<int> expected_LT_colptr_Values { 0, 0, 0, 0, 0, 0, 1, 3, 3, 3, 3, 4, 4, 5, 7, 11 };
     EXPECT_EQ(invertData->LT_colptr, expected_LT_colptr_Values);
 }
+
+TEST(SparseLDLSolver, EmptySystem)
+{
+    // required to be able to use EXPECT_MSG_NOEMIT and EXPECT_MSG_EMIT
+    sofa::helper::logging::MessageDispatcher::addHandler(sofa::testing::MainGtestMessageHandler::getInstance() ) ;
+
+    using MatrixType = sofa::linearalgebra::CompressedRowSparseMatrix<SReal>;
+
+    MatrixType matrix;
+    matrix.resize(0, 0);
+    matrix.compress();
+
+    using Solver = sofa::component::linearsolver::direct::SparseLDLSolver<MatrixType, sofa::linearalgebra::FullVector<SReal> >;
+    Solver::SPtr solver = sofa::core::objectmodel::New<Solver>();
+
+    solver->init();
+
+    EXPECT_MSG_EMIT(Warning);
+    solver->invert(matrix);
+}
+
+TEST(SparseLDLSolver, EmptyMState)
+{
+    // required to be able to use EXPECT_MSG_NOEMIT and EXPECT_MSG_EMIT
+    sofa::helper::logging::MessageDispatcher::addHandler(sofa::testing::MainGtestMessageHandler::getInstance() ) ;
```

Comment:
you need this even if you are using simpleAPI ?

Reply:
I don't see how it is related

Reply:
it is just that I never had to add manually the MessageDispatcher in the tests I wrote. So I'm wondering if the simpleapi is doing it maybe through the createNewGraph 
I will check, if I have time.

Reply:
Actually you don't have to do that if you have a test deriving from `BaseTest`

Reply:
ok, thanks for the explanation